src/Vector.cpp

```diff
@@ -0,0 +1,215 @@
+#pragma once
+#include "Vector.h"
+
+
+Vector::Vector(const size_t& size, const int& value)
+    : _size(0), _capacity(0), _data(nullptr) {
+    for (size_t i = 0; i < size; i++) {
+        push_back(value);
+    }
+}
+
+Vector::~Vector() {
+    if (_data != nullptr)
```

Comment:
This if makes sense, you don't want to call delete[] on nullptr. But it isn't required since calling delete on nullptr has no effect. So the if can be dropped for better performance.
Check it out: https://stackoverflow.com/questions/50290022/calling-delete-or-delete-on-null-pointer